src/server/replica.cc

```diff
@@ -587,6 +584,13 @@ error_code Replica::InitiateDflySync(std::optional<LastMasterSyncData> last_mast
         std::accumulate(is_full_sync.get(), is_full_sync.get() + num_df_flows, 0);
 
     if (num_full_flows == num_df_flows) {
+      // Make sure we're in LOADING state.
+      if (!service_.RequestLoadingState()) {
```

Comment:
Should be safe to do, journal changes loaded via the snapshot are executed via `DispatchCommand` and incremental snapshot only adds journal changes to the snapshot.